cvat/apps/events/export.py

```diff
@@ -0,0 +1,155 @@
+from cvat.apps.engine.log import slogger
+
+DEFAULT_CACHE_TTL = timedelta(hours=1)
+
+def _create_csv(query_params, output_filename, cache_ttl):
+    try:
+        clickhouse_settings = settings.CLICKHOUSE['events']
+
+        time_filter = {
+            'from': query_params.pop('from'),
+            'to': query_params.pop('to'),
+        }
+
+        query = "SELECT * FROM events"
+        conditions = []
+        parameters = {}
+        for param, value in query_params.items():
+            if value:
+                conditions.append(f"{param} = {{{param}:UInt64}}")
+                parameters[param] = value
+
+        if time_filter['from']:
+            conditions.append(f"timestamp >= {{from:DateTime64}}")
+            parameters['from'] = time_filter['from']
+
+        if time_filter['to']:
+            conditions.append(f"timestamp <= {{to:DateTime64}}")
+            parameters['to'] = time_filter['to']
+
+        if conditions:
+            query += " WHERE " + " AND ".join(conditions)
+
+        with clickhouse_connect.get_client(
+            host=clickhouse_settings['HOST'],
+            database=clickhouse_settings['NAME'],
+            port=clickhouse_settings['PORT'],
+            username=clickhouse_settings['USER'],
+            password=clickhouse_settings['PASSWORD'],
+        ) as client:
+            result = client.query(query, parameters=parameters)
+
+        with open(output_filename, 'w', encoding='UTF8') as f:
+            writer = csv.writer(f)
+            writer.writerow(result.column_names)
+            writer.writerows(result.result_rows)
+
+        archive_ctime = os.path.getctime(output_filename)
+        scheduler = django_rq.get_scheduler(settings.CVAT_QUEUES.IMPORT_DATA.value)
+        cleaning_job = scheduler.enqueue_in(time_delta=cache_ttl,
+            func=clear_export_cache,
+            file_path=output_filename,
+            file_ctime=archive_ctime,
+        )
+        slogger.glob.info(
+            f"The {output_filename} is created "
+            f"and available for downloading for the next {cache_ttl}. "
+            f"Export cache cleaning job is enqueued, id '{cleaning_job.id}'"
+        )
+        return output_filename
+    except Exception:
+        log_exception(slogger.glob)
+        raise
+
+def export(request, queue_name):
+    action = request.query_params.get('action', None)
+    filename = request.query_params.get('filename', None)
+
+    query_params = {
+        'organization': request.query_params.get('org', None),
+        'project': request.query_params.get('project', None),
+        'task': request.query_params.get('task', None),
+        'job': request.query_params.get('job', None),
+        'user': request.query_params.get('user', None),
+        'from': request.query_params.get('from', None),
+        'to': request.query_params.get('to', None),
+    }
+
+    if query_params['from']:
+        query_params['from'] = parser.parse(query_params['from']).timestamp()
```

Comment:
Can we have an exception here? Should we protect ourselves from a wrong user input?